### PR TITLE
Fix #13771: Refactor EditorObjectiveOptions to use Window class

### LINKED ISSUE #13771: Refactor window to class: EditorObjectiveOptions
See #13737 for more details.

### PR DESCRIPTION
**Pull request by toadkarter**
This PR refactors EditorObjectiveOptions to use the Window class. It also removes a few unused includes.

The refactored code was tested on a Windows x86 build and compared against equivalent behaviour in the latest release to ensure that all behaviour renamed unchanged. Some of the local variables were updated as the compiler was giving warnings about the names shadowing the outer scope - I tried to ensure that the meaning was as similar to the original as possible (i.e., change "width" to "currentWidth". 

I had a minor query in relation to a method in the original code called `WindowEditorObjectiveOptionsRidesScrollgetheight`. There does not seem to be a corresponding Window event for this in the same way that there is for other scroll event like `OnScrollMouseOver` or `OnScrollMouseDown`, so I wasn't sure what to do with this. Because this method doesn't seem to be called anywhere, I just renamed it in a similar convention to the other scroll events and left it in the class, but perhaps further edits are required here - if so just let me know and I will incorporate.

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
> original code called `WindowEditorObjectiveOptionsRidesScrollgetheight`. 

If it's not called by anything just delete it.

**Comment by toadkarter** [comment]
> If it's not called by anything just delete it.

Cool - thanks! That's gone now

File: src/openrct2-ui/windows/EditorObjectiveOptions.cpp [hunk 1]
```diff
@@ -191,925 +132,1019 @@ static uint64_t window_editor_objective_options_page_hold_down_widgets[] = {
...
-    WindowInitScrollWidgets(*w);
-    w->Invalidate();
-}
+    void OnMouseUp(WidgetIndex widgetIndex) override
+    {
+        switch (widgetIndex)
+        {
+            case WIDX_CLOSE:
+                Close();
+                break;
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/EditorObjectiveOptions.cpp:156]: return for this and l160. This is important as after close the class is invalid so we don't want anything else to be accessed.

**Comment by duncanspumpkin** [comment]
> > If it's not called by anything just delete it.
> 
> Cool - thanks! That's gone now

Not sure what i was thinking its `OnScrollGetSize` that it is meant to be replaced with. Could you readd it and link it to that.

**Review by duncanspumpkin — CHANGES_REQUESTED** [review]
See comments

File: src/openrct2-ui/windows/EditorObjectiveOptions.cpp (hunk 1 above)
Comment (toadkarter) [reply src/openrct2-ui/windows/EditorObjectiveOptions.cpp:156]: My bad, yeah I remember having to add something like this to my previous PR too - will include in the next push!

**Comment by toadkarter** [comment]
> > > If it's not called by anything just delete it.
> > 
> > 
> > Cool - thanks! That's gone now
> 
> Not sure what i was thinking its `OnScrollGetSize` that it is meant to be replaced with. Could you readd it and link it to that.

This one I was a little less sure of, as from looking at the codebase implementations of `OnScrollGetSize`, these seem to be a lot more expansive than what is in `EditorObjectiveOptions`, and also return a `ScreenSize` object. 

As you will see in my implementation I created a `ScreenSize` object where the height is the modified height as outlined by the original logic, and return that object. I keep the width unset because as far as I can tell this is how it works in `Guest`'s implementation of this method. 

Hopefully this approach works but if any changes need to be made just let me know and I will do so.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**